cse/src/main/java/com/adyen/checkout/cse/internal/CardEncryptorFactory.kt

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright (c) 2023 Adyen N.V.
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ *
+ * Created by oscars on 27/9/2023.
+ */
+
+package com.adyen.checkout.cse.internal
+
+import androidx.annotation.RestrictTo
+
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+object CardEncryptorFactory {
+
+    fun provide(): BaseCardEncryptor {
+        val dateGenerator = DateGenerator()
```

Comment:
Is there a way to not instantiate these dependencies here? I think previously we were doing that in the provider and using provide only in the public facing classes

Reply:
The intention here was to make `DateGenerator` and `JSONWebEncryptor` `internal` and as a side effect it cleans up the component providers a bit